include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -169,15 +171,14 @@ namespace internal
         temp2 = temp1 + temp_size;
       }
 
-    VectorizedArray<Number> **values_dofs = values_dofs_actual;
-    VectorizedArray<Number> *expanded_dof_values[n_components];
+    const unsigned int n_q_points = temp_size == 0 ? shape_info.n_q_points : Eval::n_q_points;
+    const unsigned int dofs_per_comp = (type == MatrixFreeFunctions::truncated_tensor) ?
+                                       Utilities::fixed_power<dim>(shape_info.fe_degree+1) : shape_info.dofs_per_component_on_cell;
+    Number *values_dofs = (type == MatrixFreeFunctions::truncated_tensor) ?
+                          scratch_data+2*(std::max(shape_info.dofs_per_component_on_cell, shape_info.n_q_points)) :
+                          const_cast<Number *> (values_dofs_actual);
```

Comment:
I don't like the const cast, wouldn't it be cleaner to move this variable below the if and make it const and operate on a different (non-const) variable inside the if? (this assumes this is the only place you modify it)

note: I hope I understood the code correctly.

Reply:
We need to check this carefully. I agree that it is not pretty right now.

Reply:
Fixed.